packages/vite/src/node/config.ts

```diff
@@ -369,6 +370,13 @@ export async function resolveConfig(
     }
   }
 
+  const server = config.server || {}
+  const serverRoot = path.posix.resolve(
```

Comment:
Should use `path.resolve` and normalize later, otherwise the blackslashs on Windows won't be resolved correctly.